pallets/sto/src/lib.rs

```diff
@@ -0,0 +1,398 @@
+        /// Total fundraisers created for a token
+        FundraiserCount get(fn fundraiser_count): map hasher(twox_64_concat) Ticker => u64;
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Create a new offering. A fixed amount of pre-minted tokens are put up for sale at the specified tiered rate.
+        #[weight = 800_000_000]
+        pub fn create_fundraiser(
+            origin,
+            offering_portfolio: PortfolioId,
+            offering_asset: Ticker,
+            raising_portfolio: PortfolioId,
+            raising_asset: Ticker,
+            tiers: Vec<PriceTier<T::Balance>>,
+            venue_id: u64,
+            start: Option<T::Moment>,
+            end: Option<T::Moment>,
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            ensure!(T::Asset::primary_issuance_agent(&offering_asset) == did, Error::<T>::Unauthorized);
+
+            let venue = VenueInfo::get(venue_id).ok_or(Error::<T>::InvalidVenue)?;
+            ensure!(
+                venue.creator == did && venue.venue_type == VenueType::Sto,
+                Error::<T>::InvalidVenue
+            );
+
+            ensure!(raising_portfolio.did == did, Error::<T>::InvalidPortfolio);
+            ensure!(<PortfolioAssetBalances<T>>::contains_key(raising_portfolio, raising_asset), Error::<T>::InvalidPortfolio);
+
+            ensure!(offering_portfolio.did == did, Error::<T>::InvalidPortfolio);
+            ensure!(<PortfolioAssetBalances<T>>::contains_key(offering_portfolio, offering_asset), Error::<T>::InvalidPortfolio);
+            let asset_balance = <PortfolioAssetBalances<T>>::get(offering_portfolio, offering_asset);
+
+            ensure!(
+                tiers.len() > 0 && tiers.iter().all(|t| t.total > 0.into()),
+                Error::<T>::InvalidPriceTiers
+            );
+
+            let offering_amount: T::Balance = tiers
+                .iter()
+                .map(|t| t.total)
+                .fold(0.into(), |total, x| total + x);
+            ensure!(offering_amount <= asset_balance, Error::<T>::InsufficientTokensRemaining);
+
+            let mut tiers = tiers;
+            // (Stable) sort by price.
+            tiers.sort_by_key(|a| a.price);
+
+            // TODO: Take custodial ownership of $sell_amount of $offering_token from primary issuance agent?
+            let fundraiser_id = Self::fundraiser_count(offering_asset) + 1;
+            let fundraiser = Fundraiser {
+                    offering_portfolio,
+                    offering_asset,
+                    raising_portfolio,
+                    raising_asset,
+                    tiers: tiers.into_iter().map(Into::into).collect(),
```

Comment:
Let's put a limit on the number of tiers. We need to iterate over this field so we don't want this to be of infinite length.

That being said, it's not that computationally expensive to iterate over tiers so we can have a relatively high limit.

How does 25 sound?

cc @adamdossa